### DISCUSSION TITLE
Formatting is not as expected (as in the docs) when using a class reference

### DISCUSSION THREAD

**Discussion by Luna-Klatzer** [mkdocstrings/mkdocstrings#306] [Q&A] [answered]
## Summary
I am currently facing a weird issue, where the auto-added formatted code is not correctly added/as expected in the docs examples here https://mkdocstrings.github.io/reference/handlers/base/. I noticed I am not importing a module, but simple classes alone, though I expected the formatting to be similar.

I am adding a class to my docs, but when doing so, the formatting seems to be odd, since:
- all functions are header size 2 and not nested
- no properties are being shown, despite them existing
- The overall design that was shown in the docs does not seem to apply here. 
- The class name only appears as a the direct code reference, though I wish to only display the name of the class.

**Example:**

My table of content:
![image](https://user-images.githubusercontent.com/61756091/129889382-3d9dc2b6-85de-44d1-80d8-ea48aacfc90c.png)

Expected table of content:
![image](https://user-images.githubusercontent.com/61756091/129890165-1ac895a8-b3e2-4863-911c-9e11e7c5cad2.png)

My class:
![image](https://user-images.githubusercontent.com/61756091/129889406-b6bbc5f9-15f1-4dc4-b114-38b9eaa845cb.png)

Expected class:
![image](https://user-images.githubusercontent.com/61756091/129890234-d159de4f-a82c-438c-b2aa-258cda2a3939.png)

**Comment by pawamoy** [comment, marked as answer]
Hi! You need to change a few options:

```yaml
plugins:
- mkdocstrings:
    handlers:
      python:
        selection:
          docstring_style: restructured-text
        rendering:
          show_root_heading: true
```

See https://mkdocstrings.github.io/handlers/python/#rendering and https://mkdocstrings.github.io/handlers/python/#selection.

For indentation, see https://mkdocstrings.github.io/handlers/python/#recommended-style-material.

For the properties not being shown, we'll need a reproducible example!

**Reply by Luna-Klatzer** [reply]
Thank you, that all solved my issue ^^ Though for the reproductive example, it seems to be a one-time case, since it didn't appear again for me.
